From 1b3e22dcb12d650e08316c28382dc224e9b75148 Mon Sep 17 00:00:00 2001
From: Daniel-Constantin Mierla <miconda@gmail.com>
Date: Mon, 12 Feb 2018 08:50:54 +0100
Subject: [PATCH] lcr: fixed checking if there is more uris available for
 next_gw

- manual backport for 732a3153a0a41d5f951fff85de607f0b46ae73da

(cherry picked from commit 43c9ad6f4dc2e20dd40a2d3dfa6b03f3dd79e3d4)
---
 modules/lcr/lcr_mod.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/lcr/lcr_mod.c b/modules/lcr/lcr_mod.c
index a90fb89..91a3f40 100644
--- a/modules/lcr/lcr_mod.c
+++ b/modules/lcr/lcr_mod.c
@@ -2460,8 +2460,8 @@ static int next_gw(struct sip_msg* _m, char* _s1, char* _s2)
 	 * Take Request-URI user from ruri_user_avp and generate Request
          * and Destination URIs. */
 
-	if (!generate_uris(_m, r_uri, &(ruri_user_val.s), &r_uri_len, dst_uri,
-			   &dst_uri_len, &addr, &gw_index, &flags, &tag_str)) {
+	if (generate_uris(_m, r_uri, &(ruri_user_val.s), &r_uri_len, dst_uri,
+			   &dst_uri_len, &addr, &gw_index, &flags, &tag_str) <= 0) {
 	    return -1;
 	}
     }
-- 
2.7.4

